rust-bins/src/bin/pg_test_tool.rs

```diff
@@ -0,0 +1,438 @@
+    description: &str,
+    default_path: &str,
+) -> PathBuf {
+    maybe_file_path.unwrap_or_else(|| {
+        let validator = |candidate: &String| -> Result<(), String> {
+            if std::path::Path::new(candidate).exists() {
+                Ok(())
+            } else {
+                Err(format!("File {} does not exist. Try again.", candidate))
+            }
+        };
+
+        let mut input = Input::new();
+        input.with_prompt(format!("Enter the path to the {} file", description));
+        if std::path::Path::new(default_path).exists() {
+            input.default(default_path.to_string());
+        };
+        input.validate_with(validator);
+        loop {
+            match input.interact() {
+                Ok(x) => return PathBuf::from(x),
+                Err(e) => println!("{}", e),
+            }
+        }
+    })
+}
+
+/// Read privacy guardian information from a file, determining how to parse them
+/// from the version number.
+pub fn read_pg_info<P: AsRef<Path> + Debug>(filename: P) -> std::io::Result<ArInfo<ArCurve>> {
+    let vars: Versioned<serde_json::Value> = read_json_from_file(filename)?;
+    match vars.version {
+        Version { value: 0 } => Ok(serde_json::from_value(vars.value)?),
+        other => Err(std::io::Error::new(
+            std::io::ErrorKind::InvalidData,
+            format!("Invalid privacy guardian version {}.", other),
+        )),
+    }
+}
+
+/// Encode a short string in a field element.
+///
+/// This function takes a string of length at most 31 bytes and converts
+/// it to a field element.
+///
+/// Encoding fails if the string is not ASCII or contains \x00.
+fn to_field_element<C: Curve>(m: String) -> Option<Value<C>> {
+    if !m.is_ascii() || m.contains("\x00") {
+        return None;
+    }
+    let mut buf = [0u8; 32];
+    let len = m.as_bytes().len();
+    if len > 31 {
+        return None;
+    }
+    let rev_bytes: Vec<u8> = m
+        .as_bytes()
+        .iter()
+        .filter(|x| **x != 0)
+        .copied()
+        .rev()
+        .collect();
+    buf[(32 - len - 1)..31].copy_from_slice(&rev_bytes);
+    let s = C::scalar_from_bytes(buf);
+    Some(Value::<C>::new(s))
+}
+
+/// Decode a string from a field element.
+///
+/// Zero bytes (`\x00`) are ignored.
+/// This function may have undefined behavior if the encoded string is is not
+/// ASCII or contains \x00.
+fn from_field_element<C: Curve>(v: Value<C>) -> String {
+    let v: C::Scalar = *v;
+    let mut bytes = Vec::new();
+    v.serial(&mut bytes);
+    let bytes: Vec<u8> = bytes.into_iter().skip(1).filter(|x| *x != 0).collect();
+    from_utf8(&bytes)
+        .expect("Could not convert bytes to string")
+        .to_string()
+}
+
+/// Encrypt an encoded message using a given public key.
+///
+/// Technically, this is "In-the-exponent" ElGamal encryption where the message
+/// is split into chunks.
+fn encrypt_msg<C: Curve>(m: Value<C>, pk: &PublicKey<C>, g: &C) -> [Cipher<C>; 8] {
```

Comment:
I think it could make sense to re-use `encrypt_prf_share` from account_holder.rs in rust-src/concordium_base/id